### DISCUSSION TITLE
Performances issues with big images/lot of images vs figma : maybe a desktop app would improve performances ?

### DISCUSSION THREAD

**Discussion by DesignThinkerer** [uxbox/uxbox#1482] [General]
Figma can handle more than 2000 1920x1080 pictures without issues while penpot becomes unusable around 200. 

Demo:

https://www.youtube.com/watch?v=p5R1q8qIIOU

Maybe a desktop version could help to improve performances ?

**Comment by diacritica** [comment]
Hi @TheophileDesmedt Thanks for your comment. "Performance" on a design & prototype tool is a "key feature" and we're well aware of this. Figma has set the benchmark on this, this is absolutely true. They have many other limitations but performance is not one of them. At Penpot we want to make sure "performance review" is part of the core dev process. 

OK, so to make the technical discussion more specific and useful for everyone we will add some "qualifier" to the "performance" term. That way we will be able to talk about "exporting performance", "workspace performance" (the one you'd be referring to), "network performance" (incl. real-time collab), "backend performance", etc.

Would a desktop app improve workspace performance? Most probably being able to use native OS calls might avoid the browser container and give us a boost but the reality is that web/browser technology has A LOT of room for improvement still. A desktop app using Electron wouldn't address any major performance issue and going "native" for all major operating systems (desktop and mobile) is just out of the question, too much work and porting issues.

I suggest we should do a dev live stream on this on February so we can go into more details about how we deal with performance at Penpot, what do you think?
